eval.go

```diff
@@ -32,11 +33,19 @@ import (
 
 const maxUInt16 = 0xffff
 
-func (m *FileMatch) addScore(what string, s float64, debugScore bool) {
-	if s != 0 && debugScore {
-		m.Debug += fmt.Sprintf("%s:%.2f, ", what, s)
+// addScore increments the score of the FileMatch by the computed score. If
+// debugScore is true, it also adds a debug string to the FileMatch. If raw is
+// -1, it is ignored. Otherwise, it is added to the debug string.
+func (m *FileMatch) addScore(what string, computed float64, raw float64, debugScore bool) {
+	if computed != 0 && debugScore {
+		m.Debug += fmt.Sprintf("%s:%.2f", what, computed)
+		if raw != -1 {
+			m.Debug += fmt.Sprintf("(%s), ", strconv.FormatFloat(raw, 'f', -1, 64)) // uses the smallest number of digits necessary to represent raw.
+		} else {
+			m.Debug += ", "
+		}
 	}
-	m.Score += s
+	m.Score += computed
 }
```

Comment:
alternative implementation to still aid reading stuff at the call sites.

```go
func addScore(s float64, debugScore bool, whatFmt string, whatArgs ...any) {
  if s != 0 && debugScore {
    var b strings.Builder
    fmt.Fprintf(&b, whatFmt, whatArgs...)
    fmt.Fprintf(&b, ":%.2f, ", s)
    m.Debug += b.String()
  }
  m.Score += s
}
```

What I don't like about what I propose is it was nice to have the string first. Then maybe consider something like

```go
func addScore(what string, s float64, debugScore bool, whatArgs ...any) {
  if s != 0 && debugScore {
    var b strings.Builder
    b.WriteString(what)
    for _, v := range whatArgs {
      fmt.Fprintf(&b, ":%v", v)
    }
    fmt.Fprintf(&b, ":%.2f, ", s)
    m.Debug += b.String()
  }
  m.Score += s
}
```

Reply:
`...any` will still cause an alloc. I added the closure to the call sites and updated the code to use `strings.Builder` because it does read nicer.